include/boost/math/special_functions/bessel.hpp

```diff
@@ -180,7 +180,7 @@ T cyl_bessel_i_imp(T v, T x, const Policy& pol)
       if(floor(v) == v)
       {
          T r = cyl_bessel_i_imp(v, T(-x), pol);
-         if(iround(v, pol) & 1)
+         if((iround(v, pol) & 1) == 1)
```

Comment:
Stupid question: I know compilers are super-smart these days, but has someone checked that this doesn't generate dis-optimised code?  At the assembly level `x&1` sets the necessary processor flags for the branch to happen, the comparison to 1 is completely superfluous.  A long long time ago when I started programming we used to have to worry about this ;)

Reply:
I did this to avoid pedantic warnings about implicit casts to bool. You are right though.